src/mpid/ch4/netmod/ofi/ofi_events.c

```diff
@@ -674,6 +687,9 @@ static int am_recv_event(struct fi_cq_tagged_entry *wc, MPIR_Request * rreq)
     if ((uo_msg = MPIDI_OFI_am_claim_unordered_msg(fi_src_addr, next_seqno)) != NULL) {
         am_hdr = &uo_msg->am_hdr;
         orig_buf = am_hdr;
+#if NEEDS_STRICT_ALIGNMENT
+        repeat = 1;
+#endif
         goto repeat;
```

Comment:
You have `repeat` both as a label and variable name?

The semantic is the first packet you have a header that needs alignment check and the later packets doesn't need alignment check, right? How about name the variable `need_alignment_check`?

And for label, how about `fn_repeat` to fit the mpich style?

Reply:
sounds good, I can change it

Reply:
Actually, `repeat` is a bad label name since it offers no hint of semantics -- all goto label is some kind of repeat, so we still need study the code to figure out why it is repeating. Let me look at the code again to see what is the code doing ...